src/main/haskell/kore/src/Kore/AST/Identifier.hs

```diff
@@ -119,6 +126,19 @@ instance NFData (Id level)
 instance IsString (Id level) where
     fromString = noLocationId . fromString
 
+instance Reflectable (Id level)
+  where
+    reflect id0@(Id _ _) =
+        case id0 of
+            Id {getId} ->
+                Reflect.mkStruct
+                    "Id"
+                    [ ("getId", reflect getId)
+                    -- Using the actual location is rarely useful and usually
+                    -- makes things worse.
+                    , ("idLocation", reflect AstLocationNone)
```

Comment:
We found the location to be very useful for debugging `evm-semantics`, e.g. to find the source of the axiom being applied.

Reply:
I see. In tests we rely on this being ignored when comparing, I'll try to leave the location here and solve the problem somewhere else.

Reply:
I intended to add transformations in a follow-up PR, but it seems that I need some rudiments now for dealing with id comparison.